File: python/BullsandCows.py
class Solution:
    def getHint(self, secret: str, guess: str) -> str:
        bulls, cows, d = 0, 0, {}
        for i in range(len(secret)):
            if secret[i] == guess[i]:
                bulls += 1
            else:
                if d.get(secret[i], 0) < 0:
                    cows += 1
                d[secret[i]] = d.get(secret[i], 0) + 1
                if d.get(guess[i], 0) > 0:
                    cows += 1
                d[guess[i]] = d.get(guess[i], 0) - 1
        res = "%dA%dB" % (bulls, cows)
        return res

File: C/BullsandCows.c
#include <stdio.h>
#include <stdlib.h>

#define min(a, b) (((a) < (b)) ? (a) : (b))

char* getHint(char* secret, char* guess)
{
	int hash_secret[10] = {0}, hash_guess[10] = {0};
	int bulls = 0, cows = 0;
	for (int i = 0; secret[i]; ++i)
	{
		if (secret[i] == guess[i])
			++bulls;
		else
		{
			++hash_secret[secret[i] - '0'];
			++hash_guess[guess[i] - '0'];
		}
	}
	for (int i = 0; i < 10; ++i)
		cows += min(hash_guess[i], hash_secret[i]);
	char* res = (char*)malloc(sizeof(char) * 10);
	sprintf(res, "%dA%dB", bulls, cows);
	return res;
}
